src/d/actor/d_a_obj_hami3.cpp

```diff
@@ -4,87 +4,259 @@
+            model->setAnmMtx(jntNo, *calc_mtx);
+            cMtx_copy(*calc_mtx, J3DSys::mCurrentMtx);
+        }
+    }
+    return TRUE;
     /* Nonmatching */
 }
 
 /* 0000012C-0000026C       .text CreateHeap__Q210daObjHami35Act_cFv */
 BOOL daObjHami3::Act_c::CreateHeap() {
+    J3DModelData * modelData = (J3DModelData *)dComIfG_getObjectRes(M_arcname, HAMI3_BDL_HAMI3);
+#if VERSION > VERSION_DEMO
+    JUT_ASSERT(0x71, modelData != NULL);
+#else
+    JUT_ASSERT(0x70, modelData != NULL);
+#endif
+
+
+    field_0x2D4 = mDoExt_J3DModel__create(modelData, 0, 0x11020203);
+    if (field_0x2D4 != NULL) {
+        const char* jointName;
+        JUTNameTab* jointNameTab = field_0x2D4->getModelData()->getJointName();
+        for (u16 i = 0; i < field_0x2D4->getModelData()->getJointNum(); i++){
+            jointName = jointNameTab->getName(i);
+            if (strcmp("mono1", jointName) == 0) {
+                field_0x2D4->getModelData()->getJointNodePointer(i)->setCallBack(nodeCallBack);
+                break;
+            }
+        }
+        field_0x2D4->setUserArea((u32)this);
+    } else {
+        return FALSE;
+    }
+    return TRUE;
+
+
     /* Nonmatching */
 }
 
 /* 0000026C-00000354       .text Create__Q210daObjHami35Act_cFv */
 BOOL daObjHami3::Act_c::Create() {
+    fopAcM_SetMtx(this, field_0x2D4->getBaseTRMtx());
+    
+    int switchIndex = prm_get_swSave();
+    if (fopAcM_isSwitch(this, switchIndex)) {
+        field_0x2D8 = 3;
+        field_0x2C8 = 0x4000;
+    } else {
+        field_0x2D8 = 0;
+        field_0x2C8 = 0;
+    }
+    shape_angle.x = home.angle.x + field_0x2C8;
+    init_mtx();
+
+    fopAcM_setCullSizeBox
+                (this, -1200.0f,-1200.0f,-1200.0f,1200.0f,1200.0f,1200.0f);
+    field_0x2DC = dComIfGp_evmng_getEventIdx("AMI3");
+    return TRUE;
     /* Nonmatching */
 }
 
 /* 00000354-00000450       .text Mthd_Create__Q210daObjHami35Act_cFv */
 cPhs_State daObjHami3::Act_c::Mthd_Create() {
+    fopAcM_SetupActor(this, daObjHami3::Act_c);
+    cPhs_State phase_state = dComIfG_resLoad(&field_0x2CC, M_arcname);
+    if (phase_state == cPhs_COMPLEATE_e) {
+        phase_state = MoveBGCreate(M_arcname, 7, dBgS_MoveBGProc_Typical, 0x1fc0);
+
+#if VERSION > VERSION_DEMO
+        JUT_ASSERT(0xb7, (phase_state == cPhs_COMPLEATE_e) || (phase_state == cPhs_ERROR_e));
+#else 
+        JUT_ASSERT(0xb6, (phase_state == cPhs_COMPLEATE_e) || (phase_state == cPhs_ERROR_e));
+#endif
+
+    }
+    return phase_state;
     /* Nonmatching */
 }
 
 /* 00000450-00000458       .text Delete__Q210daObjHami35Act_cFv */
 BOOL daObjHami3::Act_c::Delete() {
+    return TRUE;
     /* Nonmatching */
 }
 
 /* 00000458-000004A4       .text Mthd_Delete__Q210daObjHami35Act_cFv */
 BOOL daObjHami3::Act_c::Mthd_Delete() {
+    u32 result = MoveBGDelete();
+#if VERSION > VERSION_DEMO
+    dComIfG_resDelete(&field_0x2CC, M_arcname);
+#else
+    dComIfG_deleteObjectRes(M_arcname);
+#endif
```

Comment:
I made a macro for this version difference since it exists in most actors
```suggestion
    dComIfG_resDeleteDemo(&field_0x2CC, M_arcname);
```